### PR TITLE
[Doc] Add *HDFS2 sink connector guide*

### PR DESCRIPTION
**Pull request by Anonymitaet**
Fix https://github.com/apache/pulsar/issues/5015

### CODE REVIEW

File: site2/docs/io-hdfs2-sink.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+| `kerberosUserPrincipal` |String| false| |The principal account of Kerberos user used for authentication. |
+| `keytab` | String|false|| The full pathname of the Kerberos keytab file used for authentication. |
+| `filenamePrefix` |String| false | |The prefix of the files created inside the HDFS directory.<br/><br/>**Example**<br/> The value of topicA result in files named topicA-. |
+| `fileExtension` | String| false | | The extension added to the files written to HDFS.<br/><br/>**Example**<br/>'.txt'<br/> '.seq' |
+| `separator` | char|false | |The character used to separate records in a text file. <br/><br/>If no value is provided, the contents from all records are concatenated together in one continuous byte array. |
+| `syncInterval` | long| false || The interval between calls to flush data to HDFS disk in milliseconds. |
+| `maxPendingRecords` |int| false|Integer.MAX_VALUE |  The maximum number of records that hold in memory before acking. <br/><br/>Setting this property to 1 makes every record send to disk before the record is acked.<br/><br/>Setting this property to a higher value allows buffering records before flushing them to disk. 
+
```
Comment (Anonymitaet) [inline site2/docs/io-hdfs2-sink.md]: @tuteng the properties from line 18 - 27 does not have default values? 
(I do not find their default values in [AbstractHdfsConfig.java](https://github.com/apache/pulsar/blob/master/pulsar-io/hdfs2/src/main/java/org/apache/pulsar/io/hdfs2/AbstractHdfsConfig.java))

Comment (tuteng) [reply site2/docs/io-hdfs2-sink.md]: `syncInterval` default is 0  
`separator` `fileExtension` `filenamePrefix` `keytab` `kerberosUserPrincipal` `compression` these fields do not need to be added with default values.  
`hdfsConfigResources` `directory` these fields are required, there should be no need to add default values

Comment (Anonymitaet) [reply site2/docs/io-hdfs2-sink.md]: Got, thank you, I've incorporated.

**Comment by Anonymitaet** [comment]
run integration tests

**Comment by Anonymitaet** [comment]
run java8 tests

**Comment by Anonymitaet** [comment]
run java8 tests

**Comment by aahmed-se** [comment]
run java8 tests

**Comment by Anonymitaet** [comment]
@aahmed-se thank you

**Pull request merged**